rasa/dialogue_understanding/commands/set_slot_command.py

```diff
@@ -75,6 +76,17 @@ def run_command_on_tracker(
                     "command_executor.skip_command.slot_not_asked_for", command=self
                 )
                 return []
+        if (
+            self.name in tracker.slots
+            and isinstance(tracker.slots[self.name], CategoricalSlot)
+            and self.value not in tracker.slots[self.name].values  # type: ignore
+        ):
+            # only fill categorical slots with values that are present in the domain
+            structlogger.debug(
+                "command_executor.skip_command.categorical_slot_value_not_in_domain",
+                command=self,
+            )
+            return []
```

Comment:
I think you can extract the slot from the tracker slots beforehand, and then check the type on it. Then you don't need to use the #type: ignore comment :)

Reply:
Thank you for the suggestion thomas :D